numerapi/numerapi.py

```diff
@@ -732,6 +735,15 @@ def upload_predictions(self, file_path: str, tournament: int = 8,
         """
         self.logger.info("uploading predictions...")
 
+        # write the pandas DataFrame as a binary buffer if provided
+        buffer_csv = None
+
+        if not df is None:
+            file_path = "predictions.csv"
```

Comment:
1. not df is None => df is not None. That's the idiomatic way of writing this in Python and is easier to read
2. `file_path` is an argument of the function with a default parameter. No need to set it again here.